### PR TITLE
Setup Routing for the Views

### LINKED ISSUE #2: Set up pages
For now we have

- Homepage
- Category Page
- Product Page
- Checkout Page
- Success Page
- 404

### PR DESCRIPTION
**Pull request by SuperluminalKnight**

### CODE REVIEW

File: src/router/index.ts [hunk 1]
```diff
@@ -10,12 +10,29 @@ const router = createRouter({
...
-      // route level code-splitting
-      // this generates a separate chunk (About.[hash].js) for this route
-      // which is lazy-loaded when the route is visited.
-      component: () => import("../views/AboutView.vue")
+      path: "/category/:CategoryName",
+      name: "category",
+      component: () => import("@/views/CategoryView.vue")
+    },
+    {
+      path: "/product/:ProductId",
```
Comment (RedBlueKoi) [inline src/router/index.ts]: Shouldn't it be /category/:categoryUrl/:productUrl ?

**Comment by RedBlueKoi** [comment]
Yep, looking good. Two bonus things as nice to have:
- In JS naming convention says that variables should be named in camelCase, so would be nice to follow the guidelines and use :productId instead of :ProductId and etc. since we gonna reference them in the code
- Would be cool if you could add lang="postcss" on style tags in created components, cause if you forget to do that later on the component will not be able to process the @apply tailwind rule

**Comment by SuperluminalKnight** [comment]
Also the current setup assumes that the provided categoryName and productId exist what would happen if they don't? Would we redirect the user to the 404 page or display an error message telling that the category/product does not exist?

**Comment by RedBlueKoi** [comment]
They should get redirected to 404, but since we haven't set up content fetching yet, we gonna roll with this for now

**Review by RedBlueKoi — APPROVED** [review]

**Pull request merged**